discovery.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/vshn/exporter-filterproxy/target"
+)
+
+type targetConfigFetcher interface {
+	FetchTargetConfigs(ctx context.Context, baseTarget string, basePath string) ([]target.StaticConfig, error)
+}
+
+type multiTargetConfigFetcher map[string]targetConfigFetcher
+
+func (mf multiTargetConfigFetcher) FetchTargetConfigs(ctx context.Context, baseTarget string, basePath string) ([]target.StaticConfig, error) {
+
+	configs := []target.StaticConfig{}
+	for path, f := range mf {
+		c, err := f.FetchTargetConfigs(ctx, baseTarget, basePath+path)
+		if err != nil {
```

Comment:
Might be useful to have a `up` metric like prometheus does them? `up{downstream=asdasd}` or similar.

Reply:
It might be a good idea to expose something like that in a metrics endpoint for the filter-proxy itself. But I'd say this is out of scope for the initial implementation